lib/rubygems/dependency.rb

```diff
@@ -330,7 +330,7 @@ def to_spec
     unless prerelease?
       # Move prereleases to the end of the list for >= 0 requirements
       pre, matches = matches.partition {|spec| spec.version.prerelease? }
-      matches += pre if requirement == Gem::Requirement.default
+      matches += pre if requirement == Gem::Requirement.default || matches.empty?
```

Comment:
There's actually another existing inconsistency.

RubyGems already falls back to prereleases even when the requirement is not a prerelease requirement, as long as the requirement is specifically ">=0" (which is also the default for dependencies that don't specify any requirement).

After realizing about this, I think it'd be fine to call this a bug and fix the inconsistency.

Also, I think it can be simplified to

```suggestion
      matches = pre if matches.empty?
```

and also we can reword the comment two lines above to something like "Consider prereleases only as a fallback".

Reply:
Yes, I can see how `if requirement == Gem::Requirement.default` is a bug now.

Thanks for the explanation.